Refactor code to use the hubot-ibmcloud-objectstorage-crawler

kevan-ibm opened the pull request:
Refactor code to use the hubot-ibmcloud-objectstorage-crawler libraries for current function (no new function added)

kevan-ibm committed:
Refactor code to use the hubot-ibmcloud-objectstorage-crawler libraries for current function (no new function added)

kevan-ibm committed:
Fixed eslint error.

kurtismgit commented on src/lib/env.js:
```diff
@@ -6,68 +6,56 @@
-};
-
-// nlc service bound to application, overrides any other settings.
-if (process.env.VCAP_SERVICES && JSON.parse(process.env.VCAP_SERVICES).natural_language_classifier) {
```
We need to follow this pattern in env.js of crawler project.  This was a recent change done by @ramaboggarapu I've created an issue in our crawler project:

https://github.com/ibm-cloud-solutions/hubot-ibmcloud-objectstorage-crawler/issues/16

kevan-ibm replied:
Thanks.  That's a PR to the crawler.

kurtismgit commented on src/lib/paramHelper.js:
```diff
@@ -29,41 +29,19 @@ function ParamHelper(options) {
-ParamHelper.prototype.getMissingEnv = function() {
-	return this.missingEnv;
+ParamHelper.prototype.initializationError = function() {
+	return this.initError;
```
The init error needs to come from 'env.initError' not 'this.initError'.

kevan-ibm replied:
Fixed.  Good catch.

kurtismgit commented on src/lib/paramHelper.js:
```diff
@@ -29,41 +29,19 @@ function ParamHelper(options) {
 	this.robot = options.robot;
 	this.res = options.res;
 	this.logger = this.robot.logger;
```
doesn't look like this.logger is ever used.

kevan-ibm replied:
Removed.  Good catch.

kurtismgit commented on src/scripts/object.search.js:
```diff
@@ -96,32 +81,20 @@ module.exports = (robot, res) => {
+				if (!classifierResult.search_successful || !classifierResult.classify_result) {
+					// no result returned
+					let error = classifierResult.description ? new Error(classifierResult.description) : new Error(classifierResult);
+					context.robot.logger.error(`${TAG}: No classifiers are available at this time.`, error);
```
I was thinking that here is a good spot to emit a message to the robot response.  Essentially if search_successful is false, description is guaranteed to have a translated message that is suitable to show to the end user (within the bot client).  Without this it will be hard for the user to know why the search didn't work.

kevan-ibm replied:
Added a message.

kevan-ibm committed:
Addressed comments in PR:
1. Removed unused logger
2. Changed reference for this.initError to env.initError
3. Emit message when search fails
